### PR TITLE
[Tree] Added context menu, partially reorganized tree components.

### PR DESCRIPTION
**Pull request by nekomi2**

### CODE REVIEW

File: front-end/components/business-process/tree/tree-view.js [hunk 1]
```diff
@@ -52,55 +54,170 @@ const findExpandable = (tree) => {
...
 
-  const handleRightClick = () =>{
-    console.log('hello');
-  };
+  const renderPublishingBusinessDomains = (nodes) => (
+    <TreeItem
+      key={nodes.name}
+      nodeId={nodes.name}
+      label={nodes.name}
+      onContextMenu={(e) => handleContextMenu(e, nodes.name)}
```
Comment (lilingxi01) [inline front-end/components/business-process/tree/tree-view.js:165]: There is an issue of context menu conflict (or accidentally inherit).

After commenting out all `onContextMenu` at all tree items except EAI domain tree item, the right click menu is still existing on the child hierarchy, especially Business Process Instance, where it should have a different right click menu than the EAI Domain.

Comment (lilingxi01) [reply front-end/components/business-process/tree/tree-view.js:165]: The expected behavior should be that when we only keep the `onContextMenu` at EAI Domain hierarchy, its child nodes does not have a customized right click menu (which may inherit the parent one right now).

**Review by lilingxi01 — CHANGES_REQUESTED** [review]

File: front-end/components/business-process/tree/tree-view.js (hunk 1 above)
Comment (lilingxi01) [reply front-end/components/business-process/tree/tree-view.js:165]: @nekomi2 Potential solution: https://github.com/mui/material-ui/issues/18112

**Comment by nekomi2** [comment]
Added a mock option on the menu to show where the source is from. Idk what the behavior should be for the log instances, should they have a specific context menu as well?

**Comment by lilingxi01** [comment]
> Added a mock option on the menu to show where the source is from. Idk what the behavior should be for the log instances, should they have a specific context menu as well?

Great fix! The context menu for business process instances is pending for now. Just leave it like this and we are going to design it in the next few sprints.

**Review by lilingxi01 — APPROVED** [review]
Great work!

**Pull request merged**